lib/internal/crypto/random.js

```diff
@@ -119,6 +120,90 @@ function randomFill(buf, offset, size, cb) {
   _randomBytes(buf, offset, size, wrap);
 }
 
+// Largest integer we can read from a buffer.
+// e.g.: Buffer.from("ff".repeat(6), "hex").readUIntBE(0, 6);
+const RAND_MAX = 0xFFFF_FFFF_FFFF;
+
+// Generates an integer in [min, max) range where min is inclusive and max is
+// exclusive.
+function randomIntRange(min, max, cb) {
+  // Detect optional min syntax
+  // randomIntRange(max)
+  // randomIntRange(max, cb)
+  const minNotSpecified = typeof max === 'undefined' ||
+    typeof max === 'function';
+
+  if (minNotSpecified) {
+    cb = max;
+    max = min;
+    min = 0;
+  }
+
+  const isSync = typeof cb === 'undefined';
+  if (!isSync && typeof cb !== 'function') {
+    throw new ERR_INVALID_CALLBACK(cb);
+  }
+  if (!NumberIsSafeInteger(min)) {
+    throw new ERR_INVALID_ARG_TYPE('min', 'safe integer', min);
+  }
+  if (!NumberIsSafeInteger(max)) {
+    throw new ERR_INVALID_ARG_TYPE('max', 'safe integer', max);
+  }
+  if (!(max >= min)) {
+    throw new ERR_OUT_OF_RANGE('max', `>= ${min}`, max);
+  }
+
+  // First we generate a random int between [0..range)
+  const range = max - min;
+
+  if (!(range <= RAND_MAX)) {
+    throw new ERR_OUT_OF_RANGE(`max${minNotSpecified ? '' : ' - min'}`,
+                               `<= ${RAND_MAX}`, range);
+  }
+
+  const excess = RAND_MAX % range;
+  const randLimit = RAND_MAX - excess;
+
+  if (isSync) {
+    // Sync API
+    while (true) {
+      const x = randomBytes(6).readUIntBE(0, 6);
+      // If x > (maxVal - (maxVal % range)), we will get "modulo bias"
+      if (x > randLimit) {
+        // Try again
+        continue;
+      }
+      const n = (x % range) + min;
+      return n;
+    }
+  } else {
+    // Async API
+    const pickAttempt = () => {
+      randomBytes(6, (err, bytes) => {
+        if (err) return cb(err);
+        const x = bytes.readUIntBE(0, 6);
+        // If x > (maxVal - (maxVal % range)), we will get "modulo bias"
+        if (x > randLimit) {
+          // Try again
+          return pickAttempt();
+        }
+        const n = (x % range) + min;
+        cb(null, n);
+      });
+    };
+
+    pickAttempt();
+  }
+}
+
+// Introduce optional "min" argument eventually?
```

Comment:
If you don't want to add the optional `min` argument now, but possibly later, this should be of the form

```
// TODO(olalonde): ...
```

Reply:
But I would really consider adding `min` again :)

Reply:
Added `min` again in latest push ;)